dowhy/gcm/independence_test/ablation.py

```diff
@@ -0,0 +1,107 @@
+""" Ablation based (conditional) independence test. Testing independence via regression, i.e. if a variable has
+information about another variable, then they are dependent.
+"""
+from typing import Callable, Optional, Union, List
+
+import numpy as np
+from sklearn.kernel_approximation import Nystroem
+from sklearn.preprocessing import scale
+
+from dowhy.gcm.stats import quantile_based_fwer, estimate_f_test_p_value
+from dowhy.gcm.util.general import shape_into_2d, apply_one_hot_encoding, fit_one_hot_encoders
+
+
+def ablation_based(X: np.ndarray,
+                   Y: np.ndarray,
+                   Z: Optional[np.ndarray] = None,
+                   num_components_all_inputs: int = 40,
+                   num_runs: int = 20,
+                   p_value_adjust_func: Callable[[Union[np.ndarray, List[float]]], float] = quantile_based_fwer,
+                   f_test_samples_ratio: Optional[float] = 0.3,
+                   max_samples_per_run: int = 10000) -> float:
+    """ The main idea is that if X and Y are dependent, then X should help in predicting Y. If there is no dependency,
+    then X should not help. When Z is given, the idea remains the same, but here X and Y are conditionally independent
+    given Z if X does not help in predicting Y when knowing Z. This is, X has not additional information about Y given
+    Z. In the pairwise case (Z is not given), the performances between predicting Y based on X and predicting Y by
```

Comment:
"the performances are compared" --good to be more specific here. Is it accuracy? or L2/cross-entropy loss?

Reply:
Its the squared error, I clarified this in the docstring.